src/Microsoft.ComponentDetection.Common/DependencyGraph/ComponentRecorder.cs

```diff
@@ -12,17 +12,19 @@
 
 namespace Microsoft.ComponentDetection.Common.DependencyGraph;
 
+using Microsoft.Extensions.Logging;
+
 public class ComponentRecorder : IComponentRecorder
 {
-    private readonly ILogger log;
-
     private readonly ConcurrentBag<SingleFileComponentRecorder> singleFileRecorders = new ConcurrentBag<SingleFileComponentRecorder>();
 
     private readonly bool enableManualTrackingOfExplicitReferences;
 
-    public ComponentRecorder(ILogger log = null, bool enableManualTrackingOfExplicitReferences = true)
+    private readonly ILogger logger;
```

Comment:
I'm not going to comment on every one, but if we do go `ILogger<T>` instead of `ILogger`, make the change the whole way down.

Reply:
It should be noted that you cannot inject an `ILogger`. You must inject an `ILogger<T>`, or DI will throw an exception. This is because this class is manually instantiated.